### PR TITLE
Feature/refactoring

### LINKED ISSUE #53: Final code review/refactor

### LINKED ISSUE #57: If market has more than one schedule display and filter for it
A market can have more than one schedule object: i.e. fall/spring seasons. Make sure we are iterating over schedule to render our schedules.

### PR DESCRIPTION
**Pull request by clairefields15**
### Acceptance Criteria
Refactored everything, all files have been run through prettier for syntax/spacing, everything still works. Moved types into a separate file to clean up each component. 
Refactored so that both fetch calls were async/await and try/catch.
Added a level of error handling to fake out a 404 for the details fetch and throw an error.
Added back in mapping of schedule/days.
Error handling views are a little different than they used to be, I apologize for what it might do to testing. (We discussed on zoom)

### Issues Resolved
closes #53 
closes #57 

### Code Reviewers
@AshleyOh-bit @alexmkio @Meekb

### CODE REVIEW

File: src/components/App/App.tsx [hunk 1]
```diff
@@ -4,109 +4,57 @@ import { getData } from '../../utils/apiCalls';
...
-    }[];
-    products: string[];
-    mapsLink: string;
-    marketName: string;
-  };
-  zip: string;
-  errorCode?: string;
-}
+import ScrollToTop from '../../utils/scrollToTop';
+import { ApiMarkets, Market } from '../../types';
```
Comment (alexmkio) [inline src/components/App/App.tsx:15]: [yessssssssssss!](https://thumbs.gfycat.com/AllSpiritedEgret-size_restricted.gif)

File: src/components/App/App.tsx [hunk 2]
```diff
@@ -4,109 +4,57 @@ import { getData } from '../../utils/apiCalls';
...
-      let response = await getData(`zipSearch?zip=${zip}`);
-      let data = await checkForError(response);
+      let data = await getData(`zipSearch?zip=${zip}`);
       let cleanedData = cleanMarketsData(data.results, distance);
-      getDetails(cleanedData);
+      await getDetails(cleanedData);
       history.push('/markets');
     } catch (error) {
       setErrorCode(error.message);
-      history.push('/markets');
```
Comment (alexmkio) [inline src/components/App/App.tsx:88]: oof. how'd that get there?! Nice removal

File: src/utils/apiCalls.ts [hunk 3]
```diff
@@ -1,4 +1,7 @@
+import { checkForError } from './utils';
+
 export const getData = async (endPoint: string) => {
   let url = `https://search.ams.usda.gov/farmersmarkets/v1/data.svc/${endPoint}`;
-  return await fetch(url);
+  let response = await fetch(url);
+  return checkForError(response);
```
Comment (alexmkio) [inline src/utils/apiCalls.ts:6]: Nice! I like this here.

File: src/components/App/App.tsx [hunk 4]
```diff
@@ -4,109 +4,57 @@ import { getData } from '../../utils/apiCalls';
...
-        const data = await checkForError(response);
-        return cleanDetailsData(data.marketdetails, currentMarket.id);
-      })
-    )
-    .then(data => addScheduleToMarkets(data, filteredMarkets))
-    .then(completeData => setData(completeData))
+  const getDetails = async (filteredMarkets: ApiMarkets['markets']) => {
+    try {
+      let data = await Promise.all(
+        filteredMarkets.map(async currentMarket => {
```
Comment (alexmkio) [inline src/components/App/App.tsx:44]: I _think_ but might be wrong that async is only needed in the function declaration.

File: src/components/App/App.tsx [hunk 5]
```diff
@@ -4,109 +4,57 @@ import { getData } from '../../utils/apiCalls';
...
-      filteredMarkets.map(async currentMarket => {
-        const response = await getData(`mktDetail?id=${currentMarket.id}`);
-        const data = await checkForError(response);
-        return cleanDetailsData(data.marketdetails, currentMarket.id);
-      })
-    )
-    .then(data => addScheduleToMarkets(data, filteredMarkets))
-    .then(completeData => setData(completeData))
+  const getDetails = async (filteredMarkets: ApiMarkets['markets']) => {
+    try {
```
Comment (alexmkio) [inline src/components/App/App.tsx:42]: [wondrous](https://media0.giphy.com/media/WQx9OcM3sdvrbTy4fa/giphy.gif?cid=790b76110847ca1d96633fca23f9e391ee2d29c43efa036d&rid=giphy.gif&ct=g)

**Review by alexmkio — APPROVED** [review]

**Review by Meekb — APPROVED** [review]

File: src/components/App/App.tsx (hunk 4 above)
Comment (clairefields15) [reply src/components/App/App.tsx:44]: I just did what VSC told me to do to convert to async haha

File: src/components/Details/Details.tsx [hunk 6]
```diff
@@ -1,45 +1,28 @@
...
-  useEffect(() => {
-    if (!selectedMarket?.schedule.length) {
-      setError("Uh oh! Market not found! Please try again.")
-    } else {
-      setOpenSeason(`${selectedMarket?.schedule[0].season}`);
-      setDaysAndTimes(`${selectedMarket?.schedule[0].dayOfWeek} ${selectedMarket?.schedule[0].time}`);
-    }
-  }, [])
+  const daysAndTimes = selectedMarket?.schedule.map((sched, index) => {
+    return (
```
Comment (AshleyOh-bit) [inline src/components/Details/Details.tsx:13]: Glad we could work this back in.

File: src/components/Error/Error.tsx [hunk 7]
```diff
@@ -1,20 +1,33 @@
...
+import { ErrorProps } from '../../types';
 
+export const Error: React.FC<ErrorProps> = ({ errorCode }) => {
   return (
     <div>
-    {errorCode === 'Sorry, no markets found for that zip code. Please try again!' && <h2>Sorry, no markets found for that zip code. Please try again!</h2>}
-    { errorCode !== 'Sorry, no markets found for that zip code. Please try again!' && serverError === "500" && <h2>"Our Servers are down, please try again."</h2>}
-    { errorCode !== 'Sorry, no markets found for that zip code. Please try again!' && <h2>Something went wrong, please try again!</h2>}
+      {(errorCode === 'fake404' || errorCode === 'fakeDetails404') && (
+        <p className='fake-404-message'>
```
Comment (AshleyOh-bit) [inline src/components/Error/Error.tsx:10]: Much nicer, good job.

**Review by AshleyOh-bit — APPROVED** [review]

**Pull request merged**